pkg/encrypt/interface.go

```diff
@@ -14,31 +14,37 @@
  * limitations under the License.
  */
 
-package minio
+// Package encrypt implements a generic interface to encrypt any stream of data.
+// currently this package implements two types of encryption
+// - Symmetric encryption using AES.
+// - Asymmetric encrytion using RSA.
+package encrypt
 
-import (
-	"io"
-	"net/http"
-)
+import "io"
 
-// EncryptionMaterials - provides generic interface to encrypt
-// any stream of data. Some crypt information can be
-// save in the object metadata
-type EncryptionMaterials interface {
+// Materials - provides generic interface to encrypt any stream of data.
+type Materials interface {
```

Comment:
'Materials' could be more generic without GetIV(), GetKey() and GetMatDesc().

Think about it this way: any encryption module can save any information it wants in headers, that information will be retrieved from headers later to decrypt data.


As this becomes a separate package, I think we can replace http.Header with map[string]string.

```
type Materials interface {
   Read(b []byte) (int, error)
   SetupEncryptMode(stream io.Reader) (map[string]string, error)
   SetupDecryptMode(stream io.Reader, decryptInfo map[string]string) error
}
```